staging/src/k8s.io/cloud-provider/controllers/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -150,9 +150,37 @@ func (c *CloudNodeLifecycleController) MonitorNodes(ctx context.Context) {
 			continue
 		}
 
+		// If provider id in spec is empty it calls instanceId with node name to get provider id
+		providerID := node.Spec.ProviderID
+		if providerID == "" {
```

Comment:
This loop is only needed on  cloud.InstancesV2() is not present. I think we can go back to your original code. Sorry about that.I was just trying to see if we can remove the additional cloud api call.

Reply:
Sure no problem, after all decided to revert to the very first model